test/helpers/FakeIpcProvider.js

```diff
@@ -31,6 +31,7 @@ var FakeIpcProvider = function IpcProvider() {
     this.error = [];
     this.validation = [];
     this.notificationCallbacks = [];
+    this.connected = true;
```

Comment:
This field is checked in the RequestManager to clear subscriptions on setProvider here:

https://github.com/ethereum/web3.js/blob/5e3ad84484741e8b8a48ce2ca95e4ae499d7a157/packages/web3-core-requestmanager/src/index.js#L95-L97


It's set on the http provider on send but I can't find where/if it's set in the WS Provider (in the provider-improvements branch anyway). It's part of the TS interface. 

In the web3-js/websocket [implementation][1], it looks like connection.connected is set to true in the constructor, and unset on close. 

[1]: https://github.com/web3-js/WebSocket-Node/blob/polyfill/globalThis/lib/WebSocketConnection.js

Reply:
The ``connected`` property does exist as an object accessor [here](https://github.com/ethereum/web3.js/blob/5e3ad84484741e8b8a48ce2ca95e4ae499d7a157/packages/web3-providers-ws/src/index.js#L82) in the ``WebsocketProvider`` class.

Reply:
Ah great!  Everything looks good then.